vm/src/serde/deserialize_program.rs

```diff
@@ -163,6 +163,44 @@ pub struct Location {
     pub start_col: u32,
 }
 
+#[cfg(all(feature = "arbitrary", feature = "std"))]
+impl<'a> Arbitrary<'a> for Location {
+    fn arbitrary(u: &mut Unstructured<'a>) -> arbitrary::Result<Self> {
+        let mut locations = Vec::new();
+
+        u.arbitrary_loop(Some(0), Some(512), |u| {
+            locations.push(Location {
+                end_line: u32::arbitrary(u)?,
+                end_col: u32::arbitrary(u)?,
+                input_file: InputFile::arbitrary(u)?,
+                parent_location: None,
+                start_line: u32::arbitrary(u)?,
+                start_col: u32::arbitrary(u)?,
+            });
+            Ok(std::ops::ControlFlow::Continue(()))
+        })?;
+
+        let mut iter_location = locations.pop().unwrap_or_else(|| Location {
+            end_line: 0,
+            end_col: 0,
+            input_file: InputFile {
+                filename: "".to_string(),
+            },
+            parent_location: None,
+            start_line: 0,
+            start_col: 0,
+        });
+
+        while let Some(mut location) = locations.pop() {
+            location.parent_location = Some((Box::new(iter_location), String::arbitrary(u)?));
+            iter_location = location;
+        }
```

Comment:
This essentially converts the location vector into a linked list, right? Is this the intended behavior?

Reply:
Yes, it is. Though I want to improve the `Location` generation, I mainly wanted arbitrary to stop crashing, as it currently does with recursive structures.